aloha-core/src/test/scala/com/eharmony/aloha/semantics/LazyInitSupport.scala

```diff
@@ -0,0 +1,52 @@
+package com.eharmony.aloha.semantics
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+  * Provides support for testing that feature functions produced by semantics
+  * do not call the feature function body when optional data is not present.
+  *
+  * This should probably be mixed into unit test companion objects.
+  *
+  * @author deaktator
+  * @since 11/28/2017
+  */
+private[aloha] abstract class LazyInitSupport {
+  private val NotCalled = false
+  private val Called = true
+
+  /**
+    * Only to be modified by `wasCalled`.
+    */
+  private val indicator = new AtomicBoolean(NotCalled)
+
+  /**
+    * Attempt to set `indicator` to `true` assuming `indicator` wasn't already set.
+    * `true` if indicator wasn't previously set; otherwise, `false`.
+    */
+  private lazy val wasCalled = indicator.compareAndSet(NotCalled, Called)
```

Comment:
This was added to check that lazy vals work. If lazy vals, didn't work, that would likely be a Scala language problem, but concerns were raised about this in conversations, so I felt it prudent to add.